src/Hoc/ProtectedRoute.tsx

```diff
@@ -1,80 +1,72 @@
-import React, { useContext, useEffect, useMemo, useRef, useState } from "react";
-import { Props, UserContext } from "../Context/Context";
-import Context from "../Context/Context";
-import User from "../models/User";
-import "../App.css";
-import CircularProgress from "@mui/material/CircularProgress";
-import { useNavigate } from "react-router-dom";
-import axios from "axios";
-import config from "../config";
+import React, { useContext, useEffect, useRef} from 'react';
+import { Props, UserContext } from '../Context/Context';
+import User from '../models/User';
+import '../App.css';
+import CircularProgress from '@mui/material/CircularProgress';
+import { useNavigate } from 'react-router-dom';
+import useServiceLayer from '../serviceLayer/servicelayer';
+
+
 export default function ProtectedRoute(props: Props) {
-  const [state, setState, username] =
-    useContext<[User, Function, string|undefined, Function]>(UserContext);
-  const [loading, setLoading] = useState(true);
-  var latitude = 0;
-  var longitude = 0;
-  const navigate = useNavigate();
-  const lastSavedLatitude = useRef<number>();
-  const lastSavedLongitude = useRef<number>();
-  useEffect(() => {
-    if (localStorage.getItem("user") != undefined) {
-      setLoading(false);
-    } else {
-      setTimeout(() => {
-        navigate("/login");
-      }, 2000);
-    }
+	// eslint-disable-next-line @typescript-eslint/ban-types
```

Comment:
Dont prefer to add eslint-disable-lint until and unless no solution for the error.